probes/http/http.go

```diff
@@ -75,6 +76,8 @@ type Probe struct {
 	// statsExportInterval / p.opts.Interval. Metrics are exported when
 	// (runCnt % statsExportFrequency) == 0
 	statsExportFrequency int64
+
+	largeBodies map[*http.Request][]byte
```

Comment:
I don't think we need to keep body bytes per request. Body bytes can be shared between requests as we are not going to write to them.

We need to keep only the ReaderCloser per request as it contains the state

I'd do something like store the body bytes into probe object at the time of initialization:

p.requestBody  = []byte(p.c.GetBody())

Reply:
For some reason I thought that there can be multiple requests per probe, and that the body is chosen at the same time as request for a target is chosen.

I've updated the code to do the thing you suggested.